boxsdk/pagination/page.py

```diff
@@ -61,7 +61,7 @@ def __getitem__(self, key):
         item_class = self._translator.translate(item_json['type'])
         item = item_class(
             session=self._session,
-            object_id=item_json['id'],
+            object_id=item_json['id'] if 'id' in item_json else None,
```

Comment:
Maybe we can do something like this:

```python
item_class = self._translator.translate(item_json['type'])
kwargs = {}
if issubclass(item_class, BaseEndpoint):
    kwargs['session'] = self._session
if issubclass(item_class, BaseObject):
    kwargs['object_id'] = item_json['id'] if 'id' in item_json else None
item = item_class(response_object=item_json, **kwargs)
```

It's not great, but it should work, for now. Eventually we'll make something nicer than this for #140 .